### PR TITLE
Feat/v autocomplete issue4663 - open on items change

### LINKED ISSUE #4663: Combobox Is Not Closing When It Has Focus And :items Is Changed
### Versions and Environment
Vuetify: 1.1.5
Vue: 2.5.17
Browsers: chrome
OS: osx

### Steps to reproduce
select item 2
press button (with cursor still in combobox)

### Expected Behavior
menu should not open, and then should close

### Actual Behavior
menu opens when changing items and will not close

### Reproduction Link
https://codepen.io/Flamenco/pen/pZEqGo?editors=1111

### PR DESCRIPTION
**Pull request by dkichler**
## Description
Adds a prop for controlling the behaviour of opening the menu when the input holds focus and the `items` change.  Current behaviour is to open the menu (when not already open) in all cases.  The prop allows this behaviour to be overridden. 

## Motivation and Context
This gives the client more flexibility in controlling when the menu is invoked.  Partially fixes #4663 

Default behaviour has been maintained in this PR, but given how many developers have been reporting it as a bug, there may be merit in changing default behaviour to suppress the menu opening instead.

## How Has This Been Tested?
There are specs included in the PR demonstrating existing default behaviour, as well as (new) overridden behaviour to suppress the menu from being invoked. 

I have also tested the changes manually in a local environment.

## Markup:

<details>

Standard (will become default with the proposed solution) use case, autocomplete does not open when items change:
```vue
<v-btn label="Change contact types" @click="changesContactTypesImmediately">Change</v-btn>

<v-autocomplete :items="filteredContactNames(contactTypes)"
                :label="contactLabel"
                :item-text="contactName"
                v-model="contact">
</v-autocomplete>
```

Use case where there is an async call responsible for updating `items`, and the clients wishes for the menu to open when the call finishes and the menu holds focus:
```vue
<v-autocomplete :items="filteredContactNames(contactTypes)"
                :label="contactLabel"
                :item-text="contactName"
                open-on-items-changed
                v-model="contact">
</v-autocomplete>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
    - addresses a 'bug' issue, but I would argue is closer to new functionality.
- [x] New feature (non-breaking change which adds functionality) 
    - as per the new prop
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).  
    - I think this could be argued either way - I've targeted `dev` considering the new prop.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.
    - will wait for initial feedback on this PR

### CODE REVIEW

File: src/components/VAutocomplete/VAutocomplete.js [hunk 1]
```diff
@@ -180,6 +184,7 @@ export default {
       // User is probably async loading
       // items, try to activate the menu
       if (
+        this.openOnItemsChanged &&
```
[the remark this answers is not in the document]
Comment (dkichler) [reply src/components/VAutocomplete/VAutocomplete.js]: I believe this check is correct - the new behaviour will be to only open the menu if the user has provided a truthy `open-on-items-changed` prop.

File: test/unit/components/VAutocomplete/VAutocomplete.spec.js [hunk 2]
```diff
@@ -230,20 +230,6 @@ test('VAutocomplete.js', ({ mount, shallow, compileToFunctions }) => {
...
-    wrapper.setProps({
-      items: [
-        { text: 'Foo', value: 1 },
-        { text: 'Bar', value: 2 }
-      ]
-    })
-
-    await wrapper.vm.$nextTick()
-
-    // Once items refresh active will
```
Comment (dkichler) [inline test/unit/components/VAutocomplete/VAutocomplete.spec.js]: This check was based on the previous default behaviour of opening the menu when items changed.  It is now covered in the more explicit specs below.

**Comment by dkichler** [comment]
Ready for another look

**Comment by dkichler** [comment]
> Can the issue be fixed for both use-cases without adding another prop?

Just to be clear, the two issues being:
- default behaviour of opening the menu when focus is held and items change is not what most would expect
- when this default behaviour is invoked, the check is done in the current cycle, and is naive to the fact that the same event causing `items` to change may also cause focus to be lost on the menu (in which case the menu should not be opened)

I don't see how adding a prop can be avoided if the feature is to be maintained.  

The second issue _could_ be addressed by doing the check in the next cycle to allow for the case that focus is lost.  I've been unable to model this case properly in a unit test as it involves a race condition between when focus is lost and when the check is applied.  

I think the solution presented in the PR covers most if not all of the cases as they were reported, including the issue of the button grabbing focus, which I assume would just leave the v-autoselect configured not to open the menu all together.  If you feel there is value in doing the check in the next cycle, I can include that as well.

**Comment by johnleider** [comment]
Maybe we should start coming up with different use-case scenarios and throwing them into the markup section of this PR so that we have something tangible to go off of.

**Comment by dkichler** [comment]
I think the use cases are pretty straightforward.  Either the menu opening when items change is desirable, in which case the new prop can be used to enable this behaviour, or it is not, which is the new default behaviour.  I've updated the markup section to cover both cases.

**Comment by dkichler** [comment]
Darn, was hoping this could have been squeezed into v1.2.  What needs to be done to move this forward?  I would really appreciate this fix (as I'm sure would the many others that have reported it).

File: test/unit/components/VAutocomplete/VAutocomplete.spec.js [hunk 3]
```diff
@@ -721,25 +707,42 @@ test('VAutocomplete.js', ({ mount, shallow, compileToFunctions }) => {
     expect(input.element.value).toBe('foo')
   })
 
-  it('should show menu when items are added and hide-no-data', async () => {
```
[the remark this answers is not in the document]
Comment (KaelWD) [reply test/unit/components/VAutocomplete/VAutocomplete.spec.js]: https://github.com/vuetifyjs/vuetify/issues/4663#issuecomment-417614743

**Comment by dkichler** [comment]
Closing in favour of #5036

**Pull request closed without merging**